src/dbcp/data_mart/projects.py

```diff
@@ -540,13 +540,172 @@ def create_long_format(engine: sa.engine.Engine) -> pd.DataFrame:
     return long_format
 
 
-def _pudl_eia860m_changelog(engine: sa.engine.Engine) -> pd.DataFrame:
-    """Get the PUDL EIA860M changelog table."""
-    with engine.connect() as con:
-        pudl_eia860m_changelog = pd.read_sql_table(
-            "pudl_eia860m_changelog", con, schema="data_warehouse"
+def get_eia860m_current(
+    engine: sa.engine.Engine, date_as_of: Optional[str] = None
+) -> pd.DataFrame:
+    """Get the most recent EIA860M data."""
+    if not date_as_of:  # get most recent data
+        date_as_of = (
+            pd.read_sql(
+                "SELECT max(valid_until_date) FROM data_warehouse.pudl_eia860m_changelog",
+                engine,
+            )
+            .iat[0, 0]
+            .strftime("%Y-%m-%d")
+        )
```

Comment:
Why do you like to use SQL for the data mart tables? I feel like we should move away from storing complex SQL as python strings because it's difficult to format and read. We're also better at pandas than SQL.

Reply:
I like your point about our organizational competence being in pandas. You're probably right that most things should be in pandas vs SQL. I started using SQL here for a couple of reasons, but none are super compelling for this use case:
 
* use what users use. The final product is a SQL database. I wanted to interact with the data in the same way they would so I could be more helpful and also discover pain points first hand. But it turns out our users mostly don't use SQL so I don't think this matters here.
* development convenience. Because the data warehouse lives in a SQL database, it is faster to jump in with SQL, so I do most prototyping in SQL. I find SQL is more succinct for a lot of basic operations like subsetting, grouping, and joining. But as I got more comfortable with SQL my definition of "basic" has expanded, and probably too far.
* performance. It is more efficient to subset a table at the database rather than transferring the whole thing to pandas just to throw most of it away. But performance doesn't matter on this small data -- this is more about engineer's ego than real value 😁

Looking at these particular 860m queries, they do still mostly seem to me to be basic SELECT operations, with the possible exception of the less well known COALESCE() function.  I definitely agree that some of the longer queries should have been in pandas (like `_get_proprietary_proposed_offshore()`). But it sounds like my judgement there doesn't match yours. Where do you think we should draw the line? Should the only interaction with the database be to `pd.read_sql_table()`, not to run any queries?

Reply:
All valid points! Yeah, given the data is pretty small, reading the whole table in using `pd.read_sql_table()` feels reasonable.

I think another reason to rely on mostly pandas is it's backend agnostic. I tried to convert parts of the ETL to use SQLite and ran into issues because there are some postgres specific functions in the queries.